### PR TITLE
Implement 3D Secure 2 parameters - Tpwdcee 3668

### PR DESCRIPTION
**Pull request by rinnhofer**

### CODE REVIEW

File: Gateway/Request/AccountInfoFactory.php [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+     * $timePeriod can be specified with relative datetime formats (e.g. 'yesterday' or '-1 day')
+     *
+     * @param string $timePeriod
+     * @return int
+     * @since 2.2.0
+     */
+    private function getCustomerTransactionCountForPeriod($timePeriod)
+    {
+        //@TODO use for transactionsLastDay + Year when clarified
+        $this->orderCollection->addFieldToFilter('customer_id', $this->customerSession->getCustomerId())
```
Comment (serucee) [inline Gateway/Request/AccountInfoFactory.php]: use consts for the reused strings like customer_id

File: Gateway/Request/AccountInfoFactory.php [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+     * Create from-to date range array where from is set with datetime string
+     * $startDateStatement can be specified with relative datetime formats (e.g. 'yesterday' or '-1 day')
+     *
+     * @param string $startDateStatement
+     * @return array
+     * @since 2.2.0
+     */
+    private function getDateRangeFilter($startDateStatement)
+    {
+        $endDate = date('Y-m-d H:i:s');
```
Comment (serucee) [inline Gateway/Request/AccountInfoFactory.php]: const

File: composer.json [hunk 3]
```diff
@@ -5,18 +5,18 @@
   "minimum-stability": "beta",
   "prefer-stable": true,
   "require": {
-    "php": ">=7.0",
```
[the remark this answers is not in the document]
Comment (rinnhofer) [reply composer.json]: updated to 7.0

File: Model/Adminhtml/Source/ChallengeIndicator.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+namespace Wirecard\ElasticEngine\Model\Adminhtml\Source;
+
+use Magento\Framework\Data\OptionSourceInterface;
+use Wirecard\PaymentSdk\Constant\ChallengeInd;
+
+class ChallengeIndicator implements OptionSourceInterface
+{
+    const NO_PREFERENCE = ChallengeInd::NO_PREFERENCE;
```
[the remark this answers is not in the document]
Comment (rinnhofer) [reply Model/Adminhtml/Source/ChallengeIndicator.php]: true story

**Pull request merged**